File: ios/Doglyad/Domain/USExaminationType.swift
import Foundation

struct USExaminationType: Codable, Identifiable, Equatable {
    let id: String
    let title: [String: String]

    func getLocalizedTitle(for locale: Locale) -> LocalizedStringResource {
        let key = locale.language.languageCode?.identifier ?? "en"
        let title = title[key] ?? title.values.first ?? ""
        return LocalizedStringResource(stringLiteral: title)
    }
}

extension USExaminationType {
    static let `default`: Self = .init(
        id: "abdominalCavity",
        title: [
            "en_US": "Abdominal cavity",
            "ru_RU": "Брюшная полость"
        ]
    )
}

File: backend/app/model/us_examination_type.py
from pydantic import BaseModel


class USExaminationType(BaseModel):
    id: str
    title: dict[str, str]

    def get_localized_title(self, locale: str) -> str:
        lang = locale.split("-")[0]
        return self.title.get(lang) or next(iter(self.title.values()), "")
